hbase-testing-util/src/main/java/org/apache/hadoop/hbase/testing/TestingHBaseCluster.java

```diff
@@ -114,6 +120,41 @@
    */
   void start() throws Exception;
 
+  /**
+   * Get the address of active master if there is one.
+   */
+  Optional<ServerName> getActiveMasterAddress();
+
+  /**
+   * Get all the backup master addresses.
+   */
+  List<ServerName> getBackupMasterAddresses();
+
+  /**
+   * Get all the region server addresses.
+   */
+  List<ServerName> getRegionServerAddresses();
+
+  /**
+   * Get the server side {@link Region} interface for the specific region.
+   * <p/>
+   * This is used for CPs to test something which can only be accessed at server side, such as tags.
+   */
+  @InterfaceAudience.LimitedPrivate(HBaseInterfaceAudience.COPROC)
+  Optional<Region> getRegion(RegionInfo regionInfo);
+
+  /**
+   * Get the server side {@link OnlineRegions} interface for the specific region server.
+   * <p/>
+   * You could list the addresses of all the region server through the
+   * {@link #getRegionServerAddresses()} method.
+   * <p/>
+   * This is used for CPs to test something which can only be accessed at server side, such as tags.
+   * And also you could use the returned interface to get all regions on this region server, etc.
+   */
+  @InterfaceAudience.LimitedPrivate(HBaseInterfaceAudience.COPROC)
+  Optional<OnlineRegions> getOnlineRegionsInterface(ServerName serverName);
```

Comment:
Why I added the 'Interface' suffix is because I assumed the name 'getOnlineRegions ' is a bit confusing. It could also be explained as get the online regions, where we should return a list of Regions.
WDYT? I'm not a native English speaker so I'm fine with changing the name, if you think 'getOnlineRegions' is enough.

Reply:
I think that because the return type is Optional(OnlineRegions), the method name getOnlineRegions is clear without the Interface, and we usually don't append Interface. But it's not a big deal either way.